### PR TITLE
My Jetpack: Introduce Boost product handling class

### PR DESCRIPTION
**Pull request by leogermani**
* Introduces API endpoints to handle products
* Add tests

#### Changes proposed in this Pull Request:

* Introduces API endpoints to handle products

#### Jetpack product discussion

N/A

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Check code
* Verify tests are passing
* Does the format of the responses look good?

### CODE REVIEW

**Review by oskosk — APPROVED** [review]
Works well :shipit: 

tested with 
```
fetch( '/wp-json/my-jetpack/v1/site/products/boost', {
	credentials: 'same-origin',
	headers: {
		'X-WP-Nonce': myJetpackInitialState.apiNonce,
		'Content-type': 'application/json' }
} )
	.then( response => response.json() )
	.then( response => console.log( response) )
	.catch( error => console.log( error.responseText ) );

```
	
and
	
	
```
	fetch( '/wp-json/my-jetpack/v1/site/products/boost', {
	method:'delete',
	credentials: 'same-origin',
	headers: {
		'X-WP-Nonce': myJetpackInitialState.apiNonce,
		'Content-type': 'application/json' }
} )
	.then( response => response.json() )
	.then( response => console.log( response) )
	.catch( error => console.log( error.responseText ) );
```

File: projects/packages/my-jetpack/src/class-rest-products.php [hunk 1]
```diff
@@ -17,32 +17,77 @@ class REST_Products {
...
+			'site/products/(?P<product>[a-z\-]+)',
 			array(
-				'methods'             => \WP_REST_Server::READABLE,
-				'callback'            => __CLASS__ . '::get_product',
-				'permission_callback' => __CLASS__ . '::permissions_callback',
-				'args'                => array(
-					'product' => array(
-						'description'       => __( 'Product slug', 'jetpack-my-jetpack' ),
-						'type'              => 'string',
-						'enum'              => Products::get_product_names(),
```
Comment (retrofox) [inline projects/packages/my-jetpack/src/class-rest-products.php:39]: Why remove the `enum`? it will check the product name

Comment (leogermani) [reply projects/packages/my-jetpack/src/class-rest-products.php:39]: it's there on line 31 ;)

**Review by retrofox — APPROVED** [review]
LGTM. 🚀

**Review by oskosk — APPROVED** [review]
Tests great!

**Pull request merged**